protocol/x/clob/keeper/equity_tier_limit.go

```diff
@@ -117,48 +119,38 @@ func (k Keeper) ValidateSubaccountEquityTierLimitForNewOrder(ctx sdk.Context, or
-
-	// Include the number of stateful orders that exist outside of the `MemClob`. This includes the number of
-	// untriggered conditional orders and during `DeliverTx` we add the count of to be committed stateful orders
-	// while in `CheckTx` we add the count of uncommitted stateful orders.
+	equityTierCount := uint32(0)
 	if order.IsStatefulOrder() {
-		equityTierCount += int32(k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter))
-		if lib.IsDeliverTxMode(ctx) {
-			equityTierCount += k.GetToBeCommittedStatefulOrderCount(ctx, order.OrderId)
-		} else {
-			equityTierCount += k.GetUncommittedStatefulOrderCount(ctx, order.OrderId)
-		}
-	}
-
-	if equityTierCount < 0 {
-		if lib.IsDeliverTxMode(ctx) {
-			err = fmt.Errorf(
-				"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
-					"equityTierCount %d, memClobCount %d, (stateful order only) toBeCommittedCount %d.",
-				order,
-				equityTierCount,
-				k.MemClob.CountSubaccountOrders(ctx, subaccountId, filter),
-				k.GetToBeCommittedStatefulOrderCount(ctx, order.OrderId),
-			)
-		} else {
-			err = fmt.Errorf(
-				"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
-					"equityTierCount %d, memClobCount %d, (stateful order only) uncommittedCount %d.",
-				order,
-				equityTierCount,
-				k.MemClob.CountSubaccountOrders(ctx, subaccountId, filter),
-				k.GetUncommittedStatefulOrderCount(ctx, order.OrderId),
-			)
+		// For stateful orders we get the stateful order count which represents how many long term and
+		// triggered conditional orders exist in state. We add to that all untriggered conditional orders.
+		// If this is `CheckTx` then we must also add the number of uncommitted stateful orders that this validator
+		// is aware of (orders that are part of the mempool but have yet to proposed in a block).
+		equityTierCount = k.GetStatefulOrderCount(ctx, order.OrderId)
+		equityTierCount += k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter)
+		if !lib.IsDeliverTxMode(ctx) {
+			equityTierCountMaybeNegative := k.GetUncommittedStatefulOrderCount(ctx, order.OrderId) + int32(equityTierCount)
+			if equityTierCountMaybeNegative < 0 {
+				panic(
+					fmt.Errorf(
+						"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
+							"equityTierCount %d, statefulOrderCount %d, untriggeredSubaccountOrders %d, "+
+							"uncommittedStatefulOrderCount %d.",
+						order,
+						equityTierCountMaybeNegative,
+						k.GetStatefulOrderCount(ctx, order.OrderId),
+						k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter),
+						k.GetUncommittedStatefulOrderCount(ctx, order.OrderId),
+					),
+				)
+			}
+			equityTierCount = uint32(equityTierCountMaybeNegative)
 		}
-		panic(err)
+	} else {
+		// For short term orders we just count how many orders exist on the memclob.
+		equityTierCount = k.MemClob.CountSubaccountOrders(ctx, subaccountId, filter)
```

Comment:
Note: this doesn't account for orders that are fully matched in the operations queue, which could later get replayed then dropped (which is kind of confusing).

Is this the intended behavior?

Reply:
`placeOrder` is only invoked from `PlaceShortTermOrder` which is only invokedfrom the ante handler when its not a recheck.

So this code isn't executed on the replay path. This would mean that in some cases a user could have more open orders then what is allowed since replaying them could add too many. The user doesn't really have control over whether their short term order is replayed so I don't think its a risk.